taosii2caom2/main_app.py

```diff
@@ -125,410 +124,1034 @@
+        bp.set('Plane.calibrationLevel', CalibrationLevel.CALIBRATED)
+
+        bp.set('Artifact.productType', self.get_product_type())
+
+    def _get_data_release(self):
+        result = None
+        meta_release_value = self._get_meta_release_value()
+        if meta_release_value:
+            # JJK, ML 23-10-24 - 5 year proprietary period for the data
+            # JJK 29-10-24 - no metadata proprietary period
+            from datetime import datetime
+            from dateutil.relativedelta import relativedelta
+            x = relativedelta(years=5)
+            y = datetime.fromisoformat(meta_release_value)
+            result = x + y
+        return result
+
+    def _get_meta_release(self):
+        result = None
+        mjdref = self._lookup_for_links('HEADER/COORDSYS/MJDREF')
+        if mjdref:
+            obs_start = _to_float(self._lookup_for_links('HEADER/EXPOSURE/OBSSTART'))
+            if obs_start:
+                result = Time(mjdref, obs_start, format='mjd', scale='tt')
+        return result
+
+    def _get_meta_release_value(self):
+        result = self._get_meta_release()
+        if result:
+            result.format = 'isot'
+            result = result.value
+        return result
+
+    def _get_obs_type(self):
+        return self._lookup(f'{self._prefix}/HEADER/OBS/OBSTYPE')
+
+    def _get_provenance_inputs(self):
+        return self._lookup(f'{self._prefix}/HEADER/CALIBRATION/FILE')
+
+    def get_algorithm_name(self):
+        result = 'lightcurve'
+        temp = self._lookup(f'{self._prefix}/HEADER/OBS/EXPTYPE')
+        if temp:
+            result = temp.lower()
+        return result
+
+    def get_instrument_name(self, ext):
+        if self._storage_name.is_fsc:
+            result = 'TAOSZWOCAM'
+        else:
+            result = self._lookup(f'//FILE/ORIGIN')
+        return result
+
+    def get_observation_intent(self, ext):
+        if self._storage_name.is_lightcurve:
+            result = ObservationIntentType.SCIENCE
+        else:
+            obstype = self._get_obs_type()
+            result = ObservationIntentType.CALIBRATION
+            if obstype == 'IMAGE':
+                result = ObservationIntentType.SCIENCE
+        return result
+
+    def get_product_type(self):
+        if self._storage_name.is_lightcurve:
+            result = ProductType.SCIENCE
+        else:
+            obstype = self._get_obs_type()
+            x = {
+                'IMAGE': ProductType.SCIENCE,
+                'FOCUS': ProductType.CALIBRATION,
+                'POINTING': ProductType.CALIBRATION,
+                'POINTING_MODEL': ProductType.CALIBRATION,
+                'BIAS': ProductType.BIAS,
+                'DARK': ProductType.DARK,
+                'DOMEFLAT': ProductType.FLAT,
+                'SKYFLAT': ProductType.FLAT,
+            }
+            result = x.get(obstype) if x.get(obstype) else ProductType.AUXILIARY
```

Comment:
I think this should work

```suggestion
            result = x.get(obstype, ProductType.AUXILIARY)
```